bot/exts/halloween/spooknamerate.py

```
@@ -0,0 +1,342 @@
+                    f"`{self.bot.command_prefix}spooknamerate "
+                    "delete` to delete it, and then you can add it again"
+                )
+                return
+
+            elif data['word'] == word:
+                # Someone has added the word
+                await ctx.send("TOO LATE. Someone has already added this word.")
+                return
+
+        # otherwise
+        self.messages[message.id] = {  # store it
+            'word': word,
+            'author': message.author,
+            'score': 0,
+        }
+
+        for emoji in self.emojis_val:
+            await message.add_reaction(emoji)  # add each emoji
+
+        logger.info(f"{message.author} added the word {word!r}")
+
+        await ctx.send(f"{word!r} added successfully!\n{random.choice(self.added_messages)}")
+        return
+
+    @commands.Cog.listener()
+    async def on_reaction_add(self, reaction: discord.Reaction, user: discord.User) -> None:
+        """Ensures that each user adds one and only one reaction."""
+        try:
+            if reaction.emoji in self.emojis_val and reaction.message.id in self.messages:
+                msg = reaction.message
+                msg: Message
+                # create a custom counter
+                reaction_counter = {}
+                for r in msg.reactions:
+                    async for user in r.users():
+                        if user in reaction_counter:
+                            reaction_counter[user] += 1
+                        else:
+                            reaction_counter[user] = 1
+
+                if reaction_counter[user] > 1 and user != self.bot.user:
+                    await user.send("Sorry, you have already added a reaction, please remove your reaction and try "
+                                    "again")
+                    return await reaction.remove(user)  # removes user's reaction
+        except RuntimeError:  # The dictionary was changed in between
+            pass
```

Are you sure that's the correct excpetion, and don't silently ignore it without any explanation as to why you dont care about the exception being thrown.

---

this still stands.

---

Would rerunning the function fit in here?

---

What are you even excepting for here?

---

So basically I just saw this error pop up when I added a word right before the winner announcement. At the end of the announcement, it resets the message and when I add a reaction to a word right before the announcement, while it is iterating over the dictionary, the dictionary changes (gets reset) and raises a `RuntimeError`. At least that's what I think. I am not able to reproduce error though. So shall I delete it?

---

@thomaspet?

---

You could create an async lock, that you await before resetting. That way it wont reset until you release the lock. Which i think is better than catching a runtimeerror

---

Okay got it! 👍

---

What I mean is I think I got it. So what I am going to do is I will create a variable to indicate whether the reactions are being checked and the messages dictionary won't reset unless the variable becomes false (and I will also add this logic in when adding a word to the dictionary). Is that what you meant @thomaspet ? 😅